module.json

```diff
@@ -21,9 +21,11 @@
       "type": "Apache-2.0"
     }
   ],
-  "dependencies": {
+  "testDependencies": {
     "dlmalloc": "^1.0.0",
-    "core-util": "^1.0.0"
+    "core-util": "^1.0.0",
+    "unity": "^2.0.0",
+    "greentea-client": "^0.1.4"
```

Comment:
These should not all be test dependencies. dlmalloc and core-util are dependencies, not test dependencies.

As @0xc0170 said, please use `~` for version specifications in the `0.x.x` range. Check the yotta docs for more information: http://yottadocs.mbed.com/reference/module.html#dependencies

I don't think that `^` does the right thing in this case.